decidim-vocdoni/decidim-module-vocdoni
# Vote statistics polling on Vote step

This PR implements the polling on Admin's panel during the Vote step. 

It also introduces multiple fixes that were mentioned in other reviews: 

* Makes the Vocdoni API environment configurable:

```ruby
Decidim::Vocdoni.configure do |config|
  config.api_endpoint_env = "stg"
end
```

* Infer the Explorer Vote URL from this environment
* Opens the votes verification links in a new tab
* Adds a couple more checks on the `SetupForm`: if there's any photo on this election and if the participatory space is published 
* When the client didn't have any tokens for working with the API (after 5 Elections setup) there was a bug while calling the `collectFaucetTokens` method
* Adds a description while creating the Elections to explain what's Vocdoni (to be improved) 
* Adds the button to publish/unpublish in the Election edit form

## Screenshot

![image](https://user-images.githubusercontent.com/717367/219066020-5d0f889d-1cd8-4650-b8b7-4a6da831bed4.png)

## Testing

1. Set up an Election
2. Vote on it 
3. Run the `bin/rails decidim_vocdoni:change_election_status` task so it changes the step
4. Go to the steps
5. See that you have 1 vote
6. On another tab, vote again with other participant in census  
7. See that in less than 30 seconds you have the new vote on this page
